php_phongo.c

```diff
@@ -1591,6 +1607,178 @@ void php_phongo_populate_default_ssl_ctx(php_stream_context *ctx, zval *driverOp
 #undef SET_STRING_CTX
 } /* }}} */
 
+bool php_phongo_apply_rp_options_to_client(mongoc_client_t *client, bson_t *options TSRMLS_DC) /* {{{ */
+{
+	bson_iter_t iter;
+	mongoc_read_prefs_t *new_rp;
+	const mongoc_read_prefs_t *old_rp;
+
+	if (!(old_rp = mongoc_client_get_read_prefs(client))) {
+		MONGOC_WARNING("Client does not have a read preference.");
```

Comment:
I'm really not a fan of using MONGOC_WARNING(). I've used MONGOC_DEBUG() before for.. debugging. But this feels backwards..

It turns out php_phongo_log() doesn't actually throw exception or anything -- does even raise a php-error by default -- for _WARNING.
So something doesn't work and the user has no idea why?